wordpress-mobile/WordPress-iOS
# Update tracks to 0.9.0

In https://github.com/Automattic/Automattic-Tracks-iOS/pull/182 I added a new `device_info_app_build_configuration` property to all events.

I initially integrated this in the Woo app, which doesn't do internal betas, so I've interpreted the values in the following way:


- Debug: for local developer builds
- Alpha: for Release-alpha builds, I believe we use this for Installable Builds on PRs. Also includes `Release-Internal`
- Beta: for Release builds uploaded to TestFlight
- Production: for Release builds uploaded to the App Store

I have some doubts about categorizing internal builds because they often are released alongside TestFlight builds except through a different channel. Overall, the simplest solution was to flag them as Alpha, and the most important part of this feature was differentiating Production from the rest. The Beta group is also interesting because it goes to external contributors, but both Debug and Alpha will likely be ignored for analysis.

To test:

You can set a breakpoint in `- [TracksServiceRemote sendBatchOfEvents:withSharedProperties:completionHandler:]` and ensure that the properties argument contains Debug for device_info_app_build_configuration.

![Screen Shot 2021-06-28 at 13 59 29](https://user-images.githubusercontent.com/8739/123758353-1fdf1980-d8bf-11eb-87ca-506d97263982.png)

Check the Tracks Live View after a few minutes, and verify that the events coming from the app include the new property.

![Screen Shot 2021-06-29 at 09 55 54](https://user-images.githubusercontent.com/8739/123759481-3b96ef80-d8c0-11eb-8557-043e922aca72.png)


I haven't been able to test anything other than Debug builds yet, but I checked that the `Automattic-Tracks-iOS` target in the Pods project has the right flags:

![Screen Shot 2021-06-28 at 13 51 03](https://user-images.githubusercontent.com/8739/123758438-36857080-d8bf-11eb-80d8-58c8afdbac94.png)


## Regression Notes
1. Potential unintended areas of impact

I can't think of any, since it only adds data to existing events

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Nothing

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:
